js-extensions/src/ResourceLoadTracker.js

```diff
@@ -31,31 +32,34 @@ export default class ResourceLoadTracker {
         // Reset - for testing.
         this.pointCSSs = {};
         this.activeCSSs = {};
-    
+
         // Iterate through each plugin /jenkins-js-extension.json
         for(var i1 = 0; i1 < extensionPointList.length; i1++) {
             var pluginMetadata = extensionPointList[i1];
             var extensions = pluginMetadata.extensions; // All the extensions defined on the plugin
             var pluginCSS = pluginMetadata.extensionCSS; // The plugin CSS URL (adjunct URL).
-    
+
             // Iterate through the ExtensionPoints defined in each plugin
             for (var i2 = 0; i2 < extensions.length; i2++) {
                 var extensionPoint = extensions[i2].extensionPoint; // The extension point name.
                 var pointCSS = this.pointCSSs[extensionPoint]; // The current list of CSS URLs for the named extension point.
-    
+
                 if (!pointCSS) {
                     pointCSS = [];
                     this.pointCSSs[extensionPoint] = pointCSS;
                 }
-    
+
                 // Add the plugin CSS if it's not already in the list.
-                if (pointCSS.indexOf(pluginCSS) === -1) {
-                    pointCSS.push(pluginCSS);
+                if (pointCSS.filter((pluginCSSEntry) => pluginCSSEntry.url === pluginCSS).length === 0) {
+                    pointCSS.push({
+                        url: pluginCSS,
+                        hpiPluginId: pluginMetadata.hpiPluginId
+                    });
```

Comment:
Small nit: this filter/push approach isn't super intuitive (to me at least). Perhaps just map it to a new array ref and then assign that to `this.pointCSSs[extensionPoint]`